src/main/java/io/cdap/plugin/azure/face/extractor/AzureFaceExtractorConfig.java

```diff
@@ -0,0 +1,168 @@
+
+  @Name(SOURCE_FIELD_NAME)
+  @Description("Specifies the input field containing the binary pdf data.")
+  private final String sourceFieldName;
+
+  @Name(CONTINUE_ON_ERROR)
+  @Macro
+  @Description("Set to true if this plugin should ignore errors.")
+  private Boolean continueOnError;
+
+  @Name(FACES_SUBSCRIPTION_KEY)
+  @Description("The Azure Faces API subscription key.")
+  @Macro
+  private String facesSubscriptionKey;
+
+  @Name(EMOTION_SUBSCRIPTION_KEY)
+  @Description("The Azure emotion API subscription key.")
+  @Macro
+  private String emotionSubscriptionKey;
+
+  public AzureFaceExtractorConfig(String sourceFieldName, Boolean continueOnError,
+                                  String facesSubscriptionKey, String emotionSubscriptionKey) {
+    this.sourceFieldName = sourceFieldName;
+    this.continueOnError = continueOnError;
+    this.facesSubscriptionKey = facesSubscriptionKey;
+    this.emotionSubscriptionKey = emotionSubscriptionKey;
+  }
+
+  private AzureFaceExtractorConfig(Builder builder) {
+    sourceFieldName = builder.sourceFieldName;
+    continueOnError = builder.continueOnError;
+    facesSubscriptionKey = builder.facesSubscriptionKey;
+    emotionSubscriptionKey = builder.emotionSubscriptionKey;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(AzureFaceExtractorConfig copy) {
+    return builder()
+      .setSourceFieldName(copy.sourceFieldName)
+      .setContinueOnError(copy.continueOnError)
+      .setFacesSubscriptionKey(copy.facesSubscriptionKey)
+      .setEmotionSubscriptionKey(copy.emotionSubscriptionKey);
+  }
+
+  public String getSourceFieldName() {
+    return sourceFieldName;
+  }
+
+  public boolean getContinueOnError() {
+    return continueOnError;
+  }
+
+  public String getFacesSubscriptionKey() {
+    return facesSubscriptionKey;
+  }
+
+  public String getEmotionSubscriptionKey() {
+    return emotionSubscriptionKey;
+  }
+
+  public void validate(FailureCollector failureCollector, @Nullable Schema inputSchema) {
+    if (inputSchema == null) {
+      failureCollector.addFailure("Could not get the input schema to validate.",
+                                  "Provide a valid input schema.");
+      throw failureCollector.getOrThrowException();
+    }
+
+    Schema.Field field = inputSchema.getField(sourceFieldName);
+    if (field == null) {
+      failureCollector.addFailure(
+        String.format("Source field '%s' must be present in the input schema.", sourceFieldName), null)
+        .withConfigProperty(SOURCE_FIELD_NAME);
+      throw failureCollector.getOrThrowException();
+    }
+
+    Schema nonNullableSchema = field.getSchema().isNullable()
+      ? field.getSchema().getNonNullable()
+      : field.getSchema();
+    if (!field.getSchema().isSimpleOrNullableSimple()) {
+      failureCollector.addFailure(
+        String.format("Source field '%s' is of unexpected type.", field.getName()),
```

Comment:
`Source field '%s' is of unexpected type '%s'.`

Reply:
After all changes in validation , this check is redundant.
Removed.